automation-roles/10-validation/validate-variables-destroy/tasks/aws-validate-variables.yml

```diff
@@ -1,4 +1,12 @@
 ---
+- name: OpenShift on AWS type
+  set_fact:
+    _openshift_on_aws_type: "{{ _current_openshift_cluster.infrastructure.type }}"
+  loop: "{{ all_config.openshift | default([]) }}"
+  loop_control:
+    loop_var: _current_openshift_cluster
+  when: _current_openshift_cluster.infrastructure.type is defined
+
```

Comment:
This loop just sets the _openshift_on_aws_type variable. If there is more than 1 `openshift` definition in the yaml files, the variable will assume the value from the final definition. You need rewrite this and run the tests for every openshift definitino.